File: julia/difference-of-squares/difference-of-squares.jl
"Sum the squares of the numbers up to the given number"
square_of_sum(n::Int) = sum(1:n) ^ 2

"Square the sum of the numbers up to the given number"
sum_of_squares(n::Int) = map(n -> n ^ 2, 1:n) |> sum

"Subtract sum of squares from square of sums"
difference(n::Int) = square_of_sum(n) - sum_of_squares(n)

File: ruby/difference-of-squares/difference_of_squares.rb
class Squares
  attr_reader :range

  def initialize(num)
    @range = 1..num
  end

  def square_of_sum
    range.sum ** 2
  end

  def sum_of_squares
    range.sum{|n| n ** 2}
  end

  def difference
    square_of_sum - sum_of_squares
  end
end
